sources/tdigest.c

```diff
@@ -0,0 +1,322 @@
+#include <stdlib.h>
+#include <stdbool.h>
+#include <string.h>
+#include <math.h>
+
+#include "tdigest.h"
+
+#define M_PI 3.14159265358979323846
+
+
+typedef struct node {
+    double mean;
+    double count;
+} node_t;
+
+
+struct td_histogram {
+    // compression is a setting used to configure the size of centroids when merged.
+    double compression;
+
+    // cap is the total size of nodes
+    int cap;
+    // merged_nodes is the number of merged nodes at the front of nodes.
+    int merged_nodes;
+    // unmerged_nodes is the number of buffered nodes.
+    int unmerged_nodes;
+
+    double merged_count;
+    double unmerged_count;
+
+    node_t nodes[0];
+};
+
+static bool is_very_small(double val) {
+    return !(val > .000000001 || val < -.000000001);
+}
+
+static int cap_from_compression(double compression) {
+    return (6 * (int)(compression)) + 10;
```

Comment:
a comment what is going on here would be helpful...

Reply:
It will keep intrigued until the next PR)